Real-Dev-Squad/website-dashboard
# New UI for task accordion

### Developer Name: Joy Gupta

### PR Number(s): 606

### Description:
Currently, the UI of the task accordion on the dashboard site does not seem to be satisfactory. It lacks in terms of UI, UX, color combinations, and readability. Additionally, pagination is implemented in the frontend. 

![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/97be0344-1384-43da-8988-25026478d898)

### Requirements:
To enhance the UI and UX of the accordion, the following modifications/improvements should be implemented:

- The UI of each task card should match the one we have on the status site.
- Instead of using next and previous buttons, implement infinite scroll with pagination being handled on the backend site.
- Change the current API with the paginated one that we are using in `my-site`


### UI Fixes/Improvements:
- [x] Replace `assigned to` with `Created By`
- [x] Add `Priority` field to the right
- [x] Conditional rendering of `Progress` based on `Status`
- [x] Add `Type` field to the right
- [x] Implement infinite scroll and pagination in the backend, instead of handling it in the frontend
- [x] Use the tasks UI from my site as it also serves the same purpose
- [x] Change the current API with paginated one


### Issue Ticket Number:
- https://github.com/Real-Dev-Squad/website-dashboard/issues/601

### Backend Changes

### Frontend Changes

### Is Under Feature Flag

### Database changes: 

### Breaking changes (If your feature is breaking/missing something please mention pending tickets): 

### Deployment notes:
NA

### Tested in local

### QA Instructions, Screenshots, Recordings
Test cases added :
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/51226f83-2c38-440a-aa24-68869552fc1a)
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/83dbb3ca-50fc-4d04-adfd-49684be235a3)

Test Stats:
Run `yarn test __tests__/user-details/task-duedate-hover.test.js ` and there you can see the test case running.
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/ed22c623-4c53-400d-b7c3-71fd8259c687)

Run `yarn test` and there you can see the test case running.
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/8719887b-caa8-4f15-bf39-c21c144fa18b)


### Working Proof
#### Functionality & Responsiveness

https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/c9c4bc8f-bd0c-497f-a24f-f0e8377c7726
